src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,243 @@
+import type { Field, FieldOption } from "../data-types";
+import { Span } from "@zendeskgarden/react-typography";
+import SearchIcon from "@zendeskgarden/svg-icons/src/16/search-stroke.svg";
+import debounce from "lodash.debounce";
+import { useTranslation } from "react-i18next";
+import { EmptyValueOption } from "./EmptyValueOption";
+
+function getCustomObjectKey(targetType: string) {
+  return targetType.replace("zen:custom_object:", "");
+}
+
+const EMPTY_OPTION = {
+  value: "",
+  name: "-",
+};
+
+interface LookupFieldProps {
+  field: Field;
+  userId: number;
+  organizationId: string | null;
+  onChange: (value: string) => void;
+}
+
+export function LookupField({
+  field,
+  userId,
+  organizationId,
+  onChange,
+}: LookupFieldProps) {
+  const {
+    id: fieldId,
+    label,
+    error,
+    value,
+    name,
+    required,
+    description,
+    relationship_target_type,
+  } = field;
+  const [options, setOptions] = useState<FieldOption[]>([]);
+  const [selectedOption, setSelectedOption] = useState<FieldOption | null>(
+    null
+  );
+  const [inputValue, setInputValue] = useState<string>(value as string);
+  const [isLoadingOptions, setIsLoadingOptions] = useState<boolean>(false);
+  const { t } = useTranslation();
+
+  const customObjectKey = getCustomObjectKey(
+    relationship_target_type as string
+  );
+
+  const loadingOption = {
+    name: t(
+      "new-request-form.lookup-field.loading-options",
+      "Loading items..."
+    ),
+    id: "loading",
+  };
+
+  const noResultsOption = {
+    name: t(
+      "new-request-form.lookup-field.no-matches.found",
+      "No matches found"
+    ),
+    id: "no-results",
+  };
+
+  const fetchSelectedOption = useCallback(
+    async (selectionValue: string) => {
+      const res = await fetch(
+        `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+      );
+      const { custom_object_record } = await res.json();
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+    },
+    [customObjectKey]
+  );
+
+  const fetchOptions = useCallback(
+    async (inputValue: string) => {
+      const searchParams = new URLSearchParams();
+      searchParams.set("name", inputValue.toLocaleLowerCase());
+      searchParams.set("source", "zen:ticket");
+      searchParams.set("field_id", fieldId.toString());
+      searchParams.set("requester_id", userId.toString());
+      if (organizationId !== null)
+        searchParams.set("organization_id", organizationId);
+      setIsLoadingOptions(true);
+      try {
+        const response = await fetch(
+          `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+        );
+
+        const data = await response.json();
+        if (data !== undefined) {
```

Comment:
Is there any case when the data can be undefined? I don't think it is possible but we would probably need to check if `response.ok` is true instead

Reply:
ok I changed to check `response.ok`